types/Link.d.ts

```diff
@@ -0,0 +1,24 @@
+import { SvelteComponentTyped } from 'svelte'
+import { RouteLocation } from './Route'
+
+interface LinkProps {
+  to: string
+  replace?: boolean
+  state?: {
+    [k in string | number]: unknown
+  }
+  getProps?: (linkParams: GetPropsParams) => Record<string, any>
+}
+
+interface GetPropsParams {
+  location: RouteLocation
+  href: string
+  isPartiallyCurrent: boolean
+  isCurrent: boolean
+}
+
+type Link = SvelteComponentTyped<
+  Omit<LinkProps & svelte.JSX.HTMLProps<HTMLAnchorElement> & svelte.JSX.SapperAnchorProps, 'href'>
+>
+
+export default Link
```

Comment:
Just wondering: Is there a specific reason why the component types are exported as default? A non-default export would also work, right?

```js
export Link
```

```js
export { Link } from './Link'
```

Reply:
Yeah, sure! just wondering though, i still don't have a solid choice when it comes to exporting things, i usually mix both, is there any scenario when one export type is better than the other, or one should be preferred in almost all situations? i've read some articles recommending named exports everywhere, because they're kinda more "flexible" than their counterparts, also, default exports are just named ones called `default`, what is your opinion about that?

Reply:
Personally I prefer named everywhere, if you use it somewhere. It keeps the API consistent.

On the other hand, I like your idea where you export de component as default, and other extra data as named exports. 

Unfortunately I can't really give you a clear-cut answer. Both have their pros & cons.

Reply:
Just read this: https://github.com/TrySound/rollup-plugin-terser#why-named-export

Forgot about those arguments.

Reply:
Thank you! those sound like reasonable points to downright stop using default exports